src/contexts/SnackContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { AlertColor } from "@mui/material";
+import { createContext, ReactNode, useContext, useState } from "react";
+
+export interface Snack {
+  message: string;
+  color: AlertColor;
+  open: boolean;
+}
+
+interface ContextValue {
+  snack: Snack[];
+  addNewSnack: (snack: Snack) => void;
+  handleClose: () => void;
+}
+
+const SnackContext = createContext<ContextValue>({
+  snack: [],
+  addNewSnack(snack) {
+    console.log("error:" + snack.message);
+  },
+  handleClose: () => {
+    console.log("något gick fel");
+  },
+});
+
+interface Props {
+  children: ReactNode;
+}
+
+function SnackProvider({ children }: Props) {
+  const [snack, setSnack] = useState<Snack[]>([]);
+
+  const addNewSnack = (newSnack: Snack) => {
+    setSnack((prevState) => [...prevState, newSnack]);
+  };
+
+  const handleClose = () => {
+    setSnack((prevState) => prevState.slice(1));
+  };
```

Comment:
```suggestion
interface ContextValue {
  snack: Snack;
  addNewSnack: (snack: Snack) => void;
  handleClose: (reason?: string) => void;
}

const SnackContext = createContext<ContextValue>({
  snack: {
    message: "You forgot to add the SnackProvider",
    color: "error",
    open: true,
  },
  addNewSnack: () => {
    console.log("You forgot to add the SnackProvider");
  },
  handleClose: () => {
    console.log("något gick fel");
  },
});

interface Props {
  children: ReactNode;
}

function SnackProvider({ children }: Props) {
  const [snack, setSnack] = useState<Snack>({
    message: "",
    color: "error",
    open: false,
  });

  const handleClose = (reason?: string) => {
    if (reason === "clickaway") return;

    setSnack({ ...snack, open: false });
  };

  const addNewSnack = (snack: Snack) => {
    setSnack(snack);
  };
```

Detta är också om vi vill ändra tillbaka till att bara ha 1 snack.
Plus en liten fix på en sak jag märkte, när man hade en snack uppe och försökte öppna en till så försvann båda.